test/position-managers/Permit.t.sol

```diff
@@ -0,0 +1,277 @@
+    using PoolIdLibrary for PoolKey;
+    using StateLibrary for IPoolManager;
+
+    PoolId poolId;
+    address alice;
+    uint256 alicePK;
+    address bob;
+    uint256 bobPK;
+
+    PositionConfig config;
+
+    function setUp() public {
+        (alice, alicePK) = makeAddrAndKey("ALICE");
+        (bob, bobPK) = makeAddrAndKey("BOB");
+
+        deployFreshManagerAndRouters();
+        deployMintAndApprove2Currencies();
+
+        (key, poolId) = initPool(currency0, currency1, IHooks(address(0)), 3000, SQRT_PRICE_1_1, ZERO_BYTES);
+
+        // Requires currency0 and currency1 to be set in base Deployers contract.
+        deployAndApprovePosm(manager);
+
+        seedBalance(alice);
+        seedBalance(bob);
+
+        approvePosmFor(alice);
+        approvePosmFor(bob);
+
+        // define a reusable range
+        config = PositionConfig({poolKey: key, tickLower: -300, tickUpper: 300});
+    }
+
+    function test_domainSeparator() public view {
+        assertEq(
+            IERC721Permit(address(lpm)).DOMAIN_SEPARATOR(),
+            keccak256(
+                abi.encode(
+                    keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"),
+                    keccak256("Uniswap V4 Positions NFT"), // storage is private on EIP712.sol so we need to hardcode these
+                    keccak256("1"),
+                    block.chainid,
+                    address(lpm)
+                )
+            )
+        );
+    }
+
+    function test_permit_increaseLiquidity() public {
+        uint256 liquidityAlice = 1e18;
+        vm.prank(alice);
+        mint(config, liquidityAlice, alice, ZERO_BYTES);
+        uint256 tokenIdAlice = lpm.nextTokenId() - 1;
+
+        // alice gives bob operator permissions
+        permit(alicePK, tokenIdAlice, bob, 1);
+
+        // bob can increase liquidity on alice's token
+        uint256 newLiquidity = 2e18;
+        uint256 balance0BobBefore = currency0.balanceOf(bob);
+        uint256 balance1BobBefore = currency1.balanceOf(bob);
+        vm.startPrank(bob);
+        increaseLiquidity(tokenIdAlice, config, newLiquidity, ZERO_BYTES);
+        vm.stopPrank();
+
+        // alice's position has new liquidity
+        bytes32 positionId =
+            keccak256(abi.encodePacked(address(lpm), config.tickLower, config.tickUpper, bytes32(tokenIdAlice)));
+        (uint256 liquidity,,) = manager.getPositionInfo(config.poolKey.toId(), positionId);
+        assertEq(liquidity, liquidityAlice + newLiquidity);
+
+        // bob used his tokens to increase liquidity
+        assertGt(balance0BobBefore, currency0.balanceOf(bob));
+        assertGt(balance1BobBefore, currency1.balanceOf(bob));
+    }
+
+    function test_permit_decreaseLiquidity() public {
+        uint256 liquidityAlice = 1e18;
+        vm.prank(alice);
+        mint(config, liquidityAlice, alice, ZERO_BYTES);
+        uint256 tokenIdAlice = lpm.nextTokenId() - 1;
+
```

Comment:
can you add an expect revert bob trying to adjust alices position before the permit

Reply:
tested in `test_noPermit_decreaseLiquidityRevert`